### PR TITLE
[docs] Update installation instructions for existing React Native projects in API references

### LINKED ISSUE #24258: [docs] Broken link to "configured Expo modules"
### Summary

When I was integrating expo-image, the link in the expo-image docs "[these additional installation instructions](https://github.com/expo/expo/tree/sdk-49/packages/expo-image)." navigated me to the github readme, then noticed that the link to "configured Expo modules" is broken and it took me a while to find the proper page on the expo docs website.

Not sure, but could be a problem in other places.

But maybe the link in expo-image docs should navigate straight to https://docs.expo.dev/bare/installing-expo-modules/ instead of taking me to the github?

### Link to the related docs page

https://github.com/expo/expo/tree/sdk-49/packages/expo-image#installation

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Fixes #24258

# How

By updating the `InstallSection` component to clearly identify the following user-facing steps required to be taken for most Expo libraries when using them inside an existing React Native project (projects generated by React Native CLI).
- Add link Install Expo modules in existing React Native apps link to direct to instructions on installing the `expo` package
- Use the source URL of each library to add context about the configuration instructions for Android and iOS native projects, which are required by some of the libraries. (Most libraries require minimal steps, such as installing Pods for iOS. Some libraries also require further configuration options, which are covered by the library's README). 

# Test Plan

Run the docs locally and visit an API reference page to see these instructions, such as http://localhost:3002/versions/unversioned/sdk/cellular/.

## Preview

CleanShot 2024-07-13 at 01 35 14@2x

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/components/plugins/InstallSection.tsx [hunk 1]
```diff
@@ -16,32 +16,30 @@ const getPackageLink = (packageNames: string) =>
...
         <P>
-          If you're installing this in a <A href="/bare/overview/">bare React Native app</A>, you
-          should also follow{' '}
-          <A href={sourceCodeUrl ?? href}>
-            <DEMI>these additional installation instructions</DEMI>
-          </A>
-          .
+          If you are installing this in an <A href="/bare/overview/">existing React Native app</A>,
+          you have to <A href="/bare/installing-expo-modules/">prepare your existing project</A> to
+          use this Expo library. Then, follow the{' '}
```
Comment (Simek) [inline docs/components/plugins/InstallSection.tsx:34]:
```suggestion
          If you are installing this library in an <A href="/bare/overview/">existing React Native app</A>,
          you have to <A href="/bare/installing-expo-modules/">prepare your existing project</A> to
          use this Expo libraries first. Then, follow the{' '}
```

Wording tweak suggestion.

File: docs/components/plugins/InstallSection.tsx [hunk 2]
```diff
@@ -16,32 +16,30 @@ const getPackageLink = (packageNames: string) =>
...
+          <A href={sourceCodeUrl ?? href}>additional configuration instructions</A> to configure
+          native projects (Android and iOS) as mentioned by library's README{' '}
```
Comment (Simek) [inline docs/components/plugins/InstallSection.tsx:36]:
```suggestion
          <A href={sourceCodeUrl ?? href}>additional instructions</A> to configure
          native projects (Android and iOS) as mentioned by library's README{' '}
```

We can avoid the word repetition.

File: docs/components/plugins/InstallSection.tsx [hunk 3]
```diff
@@ -16,32 +16,30 @@ const getPackageLink = (packageNames: string) =>
...
+          <DEMI>under installation in bare React Native projects section</DEMI>.
```
Comment (Simek) [inline docs/components/plugins/InstallSection.tsx:37]:
```suggestion
          under <DEMI>"Installation in bare React Native projects"</DEMI> section.
```

Formatting suggestion.

**Review by Simek — APPROVED** [review]
Thanks for the update! 👍  

Left few minor suggestions, but we can also ship as-is too.

**Comment by amandeepmittal** [comment]
@Simek, I made some more tweaks based on your feedback and also to omit mentioning native projects part (since sometimes, there's no configuration required for Android and iOS projects, only pod install for iOS).

Also, I think mentioning "bare workflow" here next to "existing React Native app" might make more sense since that is the term most users are familiar. Not sure if completely omitting it will convey the same message. 

Here is the update after the recent changes:

CleanShot 2024-07-15 at 02 13 53

**Review by Simek — APPROVED** [review]
Looks good, thanks for the follow up! 👍

**Pull request merged**